legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalSinkCleanerAbstract.java

```diff
@@ -0,0 +1,299 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.persistence.components.relational.api;
+
+import org.finos.legend.engine.persistence.components.executor.Executor;
+import org.finos.legend.engine.persistence.components.logicalplan.LogicalPlan;
+import org.finos.legend.engine.persistence.components.logicalplan.conditions.Condition;
+import org.finos.legend.engine.persistence.components.logicalplan.conditions.Equals;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.Dataset;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.DatasetReference;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.Selection;
+import org.finos.legend.engine.persistence.components.logicalplan.operations.*;
+import org.finos.legend.engine.persistence.components.logicalplan.values.*;
+import org.finos.legend.engine.persistence.components.relational.RelationalSink;
+import org.finos.legend.engine.persistence.components.relational.SqlPlan;
+import org.finos.legend.engine.persistence.components.relational.sql.TabularData;
+import org.finos.legend.engine.persistence.components.relational.sqldom.SqlGen;
+import org.finos.legend.engine.persistence.components.relational.transformer.RelationalTransformer;
+import org.finos.legend.engine.persistence.components.transformer.TransformOptions;
+import org.finos.legend.engine.persistence.components.transformer.Transformer;
+import org.finos.legend.engine.persistence.components.util.LockInfoDataset;
+import org.finos.legend.engine.persistence.components.util.LockInfoUtils;
+import org.finos.legend.engine.persistence.components.util.MetadataDataset;
+import org.finos.legend.engine.persistence.components.util.SinkCleanupAuditDataset;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Default;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Value.Immutable
+@Value.Style(
+        typeAbstract = "*Abstract",
+        typeImmutable = "*",
+        jdkOnly = true,
+        optionalAcceptNullable = true,
+        strictBuilder = true
+)
+
+
+public abstract class RelationalSinkCleanerAbstract
+{
+    //---------- FIELDS ----------
+    public abstract RelationalSink relationalSink();
+
+    public abstract Dataset mainDataset();
+
+    public abstract Optional<LockInfoDataset> lockInfoDataset();
+
+    public abstract String requestedBy();
+
+    @Default
+    public SinkCleanupAuditDataset auditDataset()
+    {
+        return SinkCleanupAuditDataset.builder().build();
+    }
+
+    public abstract MetadataDataset metadataDataset();
+
+    @Default
+    public Clock executionTimestampClock()
+    {
+        return Clock.systemUTC();
+    }
+
+    @Default
+    public boolean enableConcurrentSafety()
```

Comment:
I looked at the flow -- we do not need to acquire the lock.
Even if we have 2 concurrent requests to clean up the sink, both of them will succeed without interfering each other.
And if we want this lock to safeguard against a running ingestion , it will not as it is not helping to avoid dropping the main table.

Reply:
Sure, removing concurrency/locks from the flow